contracts/SP1_SpokePool.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "./SpokePool.sol";
+
+interface IHelios {
+    function executionStateRoots(uint256) external view returns (bytes32);
+}
+
+interface ISP1Verifier {
+    function verifyProof(
+        bytes32 programVKey,
+        bytes calldata publicValues,
+        bytes calldata proofBytes
+    ) external view;
+}
+
+/**
+ * @notice SP1 Spoke pool capable of receiving data stored in L1 state via SP1 + Helios light clients.
+ */
+contract SP1_SpokePool is SpokePool {
+    // The public values stored on L1 that can be relayed into this contract when accompanied with an SP1 proof.
+    struct ContractPublicValues {
+        bytes32 stateRoot;
+        bytes contractCalldata;
+        bytes contractOutput;
+    }
+    /// @notice The address of the HubPool contract. Checked against public values to ensure only state
+    /// stored by HubPool is relayed.
+    address public immutable hubPool;
+
+    /// @notice The address of the SP1 verifier contract.
+    address public verifier;
+
+    /// @notice The address of the Helios light client contract.
+    address public helios;
+
+    /// @notice The verification key for the acrossCall program.
+    bytes32 public acrossCallProgramVKey;
```

Comment:
Should these be immutable? Not a leading question -- just leaving it here to make sure we consider whether we need these to be updatable without a redeploy.